jps/model-impl/src/com/intellij/openapi/fileTypes/impl/FileTypeAssocTable.java

```diff
@@ -14,17 +14,26 @@
 
 import java.util.*;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 @ApiStatus.Internal
 public final class FileTypeAssocTable<T> {
+  // Comparator to sort wildcard mappings from the longest to shortest matcher for IJPL-149806.
+  // NOTE: this is not perfect as sometimes wildcard matchers of the same length can still be considered to be of different "specificity"
+  //       (e.g. you'd probably want `*.pdf` to take precedence over `*.p?f`), but is still an improvement on the status quo.
+  private static final Comparator<Pair<FileNameMatcher, ?>>
+    MY_MATCHING_MAPPINGS_COMPARATOR = Comparator.comparing((Pair<FileNameMatcher, ?> pair) -> pair.first.getPresentableString()).reversed();
```

Comment:
please rename to meaningful name

Reply:
Saying "it's a comparator for matching mappings" seems meaningful enough to me? Unless you want the name to say something about how it sorts? So maybe something like `MY_LONGEST_MATCHING_MAPPING_FIRST_COMPARATOR` is what you have in mind?

Reply:
now, that we have only one usage of this constant, could you please inline it to a local var closer to the usage, and add there this comment above? I don't think anyone would use that constant ever more, so it doesn't make sense to have this comparator as a separate constant. the local var name can be "byNameLengthComparator" or smth similar.